pkg/trace/transform/transform.go

```diff
@@ -121,7 +121,7 @@ func OtelSpanToDDSpan(
 
 	// TODO(songy23): use AttributeDeploymentEnvironmentName once collector version upgrade is unblocked
 	if _, ok := ddspan.Meta["env"]; !ok {
-		if env := traceutil.GetOTelAttrValInResAndSpanAttrs(otelspan, otelres, true, "deployment.environment.name", semconv.AttributeDeploymentEnvironment); env != "" {
+		if env := traceutil.GetOTelAttrVal(otelres.Attributes(), true, "deployment.environment.name", semconv.AttributeDeploymentEnvironment); env != "" {
```

Comment:
If you change here, please change https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/stats/otel_util.go#L59 as well

Reply:
or better yet create a new func `GetOTelEnv` and use in both locations